### PR TITLE
Add possibility to enable exit code writing to a file

### LINKED ISSUE #48: Print exit code of the tests process to file
In order to simplify tests status processing in CI environments, we need to introduce the possibility to enable exit code reporting to a file for java process of tasks `runStories` and `debugStories`

### PR DESCRIPTION
**Pull request by ikalinin1**
Closes #48

### CODE REVIEW

File: 1.0/gradle/vividus-test.gradle [hunk 1]
```diff
@@ -123,6 +123,9 @@ check.dependsOn testVividusInitialization
 class StoriesExec extends VividusJavaExec {
     @Input
     boolean treatKnownIssuesOnlyAsPassed
+
+    @Input
+    boolean writeExitCodeToFile;
```
Comment (valfirst) [inline 1.0/gradle/vividus-test.gradle:128]:
```suggestion
    File fileToSaveExitCode;
```

**Review by valfirst — CHANGES_REQUESTED** [review]

File: 1.0/gradle/vividus-test.gradle [hunk 2]
```diff
@@ -134,6 +141,14 @@ tasks.withType(StoriesExec).configureEach {
     }
     ignoreExitValue true
     doLast {
+        def pathToExitCodeFile = project.hasProperty('fileToSaveExitCode') ? new groovy.text.SimpleTemplateEngine().createTemplate(project.fileToSaveExitCode)
+                                                                                                                   .make(project.getProperties())
+                                                                                                                   .toString()
```
Comment (valfirst) [inline 1.0/gradle/vividus-test.gradle:146]: - why not to use [`findProperty`](https://discuss.gradle.org/t/getproperty-vs-findproperty/21822/2)? 
- why is all logic around templating needed?

File: 1.0/gradle/vividus-test.gradle [hunk 3]
```diff
@@ -134,6 +141,14 @@ tasks.withType(StoriesExec).configureEach {
...
+                                                                           : fileToSaveExitCode
+        if (pathToExitCodeFile) {
+            logger.warn("Saving exit code to a file: ${pathToExitCodeFile}")
```
Comment (valfirst) [inline 1.0/gradle/vividus-test.gradle:149]: isn't `info` enough?

Comment (valfirst) [inline 1.0/gradle/vividus-test.gradle:149]: it's better to do logging after file write operation

**Review by valfirst — CHANGES_REQUESTED** [review]

File: 1.0/gradle/vividus-test.gradle (hunk 2 above)
Comment (ikalinin1) [reply 1.0/gradle/vividus-test.gradle:146]: `why not to use findProperty?
why is all logic around templating needed?`
to resolve a references
./gradlew dS  -PfileToSaveExitCode='${project.buildDir}/exitCode.txt' -p vividus-tests

File: 1.0/gradle/vividus-test.gradle (hunk 3 above)
Comment (ikalinin1) [reply 1.0/gradle/vividus-test.gradle:149]: image
 info is not printed by default, still changed to lifecycle

File: 1.0/gradle/vividus-test.gradle (hunk 2 above)
Comment (valfirst) [reply 1.0/gradle/vividus-test.gradle:146]: we shouldn't disclose gradle runtime properties in the input parameters, add an extra flag:
```
./gradlew dS -PfileToSaveExitCode=/exitCode.txt -PresolvePathAgainstProjectBuildDir=true -p vividus-tests
```

File: 1.0/gradle/vividus-test.gradle (hunk 3 above)
Comment (valfirst) [reply 1.0/gradle/vividus-test.gradle:149]: >  still changed to lifecycle

are changes lost?

**Review by valfirst — CHANGES_REQUESTED** [review]

File: 1.0/gradle/vividus-test.gradle [hunk 4]
```diff
@@ -134,6 +144,14 @@ tasks.withType(StoriesExec).configureEach {
     }
     ignoreExitValue true
     doLast {
+        def pathToExitCodeFile = project.findProperty('fileToSaveExitCode') ?: fileToSaveExitCode
+        if (pathToExitCodeFile) {
+            def pathToFile = (project.findProperty('resolvePathAgainstProjectBuildDir') ?: resolvePathAgainstProjectBuildDir)
+                    ? Paths.get("${project.buildDir}", pathToExitCodeFile)
+                    : Paths.get(pathToExitCodeFile)
+            pathToFile.toFile().write("${executionResult.get().getExitValue()}")
+            logger.lifecycle("Saving exit code to a file: ${pathToFile}")
```
Comment (valfirst) [inline 1.0/gradle/vividus-test.gradle:153]:
```suggestion
            logger.lifecycle("Exit code is saved to the file: ${pathToFile}")
```

**Pull request merged**